File: Array/PassObjectAndReturn.py
# Class definition
class Box:
    def __init__(self, width, height):
        # Initialize the Box object with width and height
        self.width = width
        self.height = height

    # Method that takes a Box object and returns a new Box object
    def create_bigger_box(self, original_box):
        # Create a new Box object with doubled dimensions
        bigger_box = Box(original_box.width * 2, original_box.height * 2)
        return bigger_box  # Return the new Box object


# Main code
# Declare and initialize the original small Box object
small_box = Box(5, 10)  # Box with width 5 and height 10

# Call the method to create a bigger box by passing small_box
bigger_box = small_box.create_bigger_box(small_box)

# Display the dimensions of the bigger Box
print(f"Bigger Box Width: {bigger_box.width}")  # Output: 10
print(f"Bigger Box Height: {bigger_box.height}")  # Output: 20

File: PassObjectAndReturn.java
// Class definition
class Box {
    int width;
    int height;

    // Method that takes a Box object as input and returns a new Box object
    Box createBiggerBox(Box originalBox) {
        // Create a new Box object
        Box biggerBox = new Box(); // Declare and initialize new object on separate lines
        biggerBox.width = originalBox.width * 2; // Doubling the width of the new Box
        biggerBox.height = originalBox.height * 2; // Doubling the height of the new Box
        return biggerBox; // Returning the new object
    }
}

public class PassObjectAndReturn {
    public static void main(String[] args) {
        // Declare a Box object
        Box smallBox;

        // Initialize the Box object
        smallBox = new Box();
        smallBox.width = 5; // Set the width of the smallBox
        smallBox.height = 10; // Set the height of the smallBox

        // Create another object to work with the method
        Box biggerBox;

        // Call the method to create a bigger Box by passing the smallBox
        biggerBox = smallBox.createBiggerBox(smallBox); // Method call to return a bigger box

        // Display the dimensions of the bigger Box
        System.out.println("Bigger Box Width: " + biggerBox.width); // Output: 10
        System.out.println("Bigger Box Height: " + biggerBox.height); // Output: 20
    }
}
